pkg/configurations/configurations.go

```diff
@@ -0,0 +1,147 @@
+package configurations
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+
+	"github.com/nmstate/kubernetes-nmstate/pkg/helper"
+	"github.com/thoas/go-funk"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	nnscontroller "github.com/nmstate/kubernetes-nmstate/pkg/controller/nodenetworkstate"
+	"github.com/spf13/viper"
+)
+
+var log = logf.Log.WithName("configurations")
+
+const defaultConfigPath = "/etc/config/"
+const defaultConfigName = "nmstate-config.yaml"
+const defaultInterfacesFilter = "veth*"
+const defaultNodeNetworkStateRefresh = "5"
+
+var (
+	configPath = defaultConfigPath
+	configName = defaultConfigName
+)
+
+// Take os environment values if given else use defaults
+func init() {
+	configPathTemp, isSet := os.LookupEnv("CONFIG_PATH")
+	if isSet {
+		configPath = configPathTemp
+	}
+	configNameTemp, isSet := os.LookupEnv("CONFIG_NAME")
+	if isSet {
+		configName = configNameTemp
+	}
+}
+
+// Struct to match config file
+type Config struct {
+	NodeNetworkRefreshInterval string `mapstructure:"node_network_state_refresh_interval"`
```

Comment:
You suffixed the config with .yaml, but here you don't treat it as a yaml. Please let's start with writting docs with examples how to set the config and figure out there what would be the simplest option for users (edit the parameter with a single kubectl command).

Reply:
Mapstructure will treat is yaml. Viper will handle it. This is the way they describe using there code.

Reply:
Same way we can have json or toml . Just took yaml as default.